src/engine/shared/demo.cpp

```diff
@@ -847,3 +859,74 @@ int CDemoPlayer::GetDemoType() const
 		return m_DemoType;
 	return DEMOTYPE_INVALID;
 }
+
+void CDemoEditor::Init(const char *pNetVersion, class CSnapshotDelta *pSnapshotDelta, class IConsole *pConsole, class IStorage *pStorage)
+{
+	m_pNetVersion = pNetVersion;
+	m_pSnapshotDelta = pSnapshotDelta;
+	m_pConsole = pConsole;
+	m_pStorage = pStorage;
+}
+
+void CDemoEditor::Slice(class CConfig *pConfig, const char *pDemo, const char *pDst, int StartTick, int EndTick)
+{
+	class CDemoPlayer DemoPlayer(m_pSnapshotDelta);
+	class CDemoRecorder DemoRecorder(m_pSnapshotDelta);
+
+	m_pDemoPlayer = &DemoPlayer;
+	m_pDemoRecorder = &DemoRecorder;
+
+	m_pDemoPlayer->SetListener(this);
+
+	m_SliceFrom = StartTick;
+	m_SliceTo = EndTick;
+	m_Stop = false;
+
+	const char *pError = m_pDemoPlayer->Load(pConfig, m_pStorage, m_pConsole, pDemo, IStorage::TYPE_ALL, m_pNetVersion);
+	if (pError)
+		return;
+
+	const CDemoPlayer::CMapInfo *pMapInfo = m_pDemoPlayer->GetMapInfo();
+	SHA256_DIGEST Fake;
+	for(unsigned i = 0; i < sizeof(Fake.data); i++)
+	{
+		Fake.data[i] = 0xff;
+	}
+	if(m_pDemoRecorder->Start(m_pStorage, m_pConsole, pDst, m_pNetVersion, pMapInfo->m_aName, Fake, pMapInfo->m_Crc, "client", true) == -1)
+		return;
```

Comment:
Instead of working around all the checks, it might be better to somehow copy the map data directly from the source demo.